src/sonic-dhcp-server/dhcp_server/common/dhcp_db_monitor.py

```diff
@@ -90,42 +125,128 @@ def _check_dhcp_server_update(self, enabled_dhcp_interfaces):
+        self.subscribe_vlan_table = swsscommon.SubscriberStateTable(self.db_connector.config_db, VLAN)
+        self.subscribe_vlan_member_table = swsscommon.SubscriberStateTable(self.db_connector.config_db, VLAN_MEMBER)
+        self.subscribe_vlan_intf_table = swsscommon.SubscriberStateTable(self.db_connector.config_db, VLAN_INTERFACE)
+        # Subscribe dhcp_server and vlan related tables.
+        self.sel.addSelectable(self.subscribe_dhcp_server_table)
+        self.sel.addSelectable(self.subscribe_dhcp_server_port_table)
+        self.sel.addSelectable(self.subscribe_dhcp_server_range_table)
+        self.sel.addSelectable(self.subscribe_dhcp_server_options_table)
+        self.sel.addSelectable(self.subscribe_vlan_table)
+        self.sel.addSelectable(self.subscribe_vlan_member_table)
+        self.sel.addSelectable(self.subscribe_vlan_intf_table)
 
-    def _check_vlan_intf_update(self, enabled_dhcp_interfaces):
+    def check_db_update(self, enabled_dhcp_interfaces, used_range, used_options):
         """
-        Check vlan_interface table
+        Fetch db and check update
         Args:
-            enabled_dhcp_interfaces: DHCP interface that enabled dhcp_server
+            enabled_dhcp_interfaces: enabled DHCP interfaces
+            used_range: used range
+            used_options: used DHCP options
         Returns:
-            Whether need to refresh
+            whether need to re-generate configuration of kea-dhcp-server
         """
-        need_refresh = False
-        while self.subscribe_vlan_intf_table.hasData():
-            key, _, _ = self.subscribe_vlan_intf_table.pop()
-            splits = key.split("|")
-            vlan_name = splits[0]
-            ip_address = splits[1].split("/")[0] if len(splits) > 1 else None
-            if vlan_name not in enabled_dhcp_interfaces:
-                continue
-            if ip_address is None or ipaddress.ip_address(ip_address).version != 4:
-                continue
-            need_refresh = True
+        state, _ = self.sel.select(self.select_timeout)
+        if state == swsscommon.Select.TIMEOUT or state != swsscommon.Select.OBJECT:
+            return False
+        need_refresh = self._check_dhcp_server_update(enabled_dhcp_interfaces)
+        need_refresh |= self.db_event_executor.check_vlan_update(enabled_dhcp_interfaces, self.subscribe_vlan_table)
+        need_refresh |= self.db_event_executor.check_vlan_intf_update(enabled_dhcp_interfaces,
+                                                                      self.subscribe_vlan_intf_table)
+        need_refresh |= self._check_dhcp_server_port_update(enabled_dhcp_interfaces)
+        need_refresh |= self._check_dhcp_server_range_update(used_range)
+        need_refresh |= self._check_dhcp_server_option_update(used_options)
+        need_refresh |= self._check_vlan_member_update(enabled_dhcp_interfaces)
         return need_refresh
+
+    def _check_dhcp_server_update(self, enabled_dhcp_interfaces):
```

Comment:
Why these methods not in _DbEventExecutor?

Reply:
Fixed